### PR TITLE
Adding well styles and basic documentation

### PR DESCRIPTION
**Pull request by jimmynotjim**
Moving the Wells organism from cfgov-refresh to Capital Framework as part of [this Design Manual task](https://github.com/cfpb/design-manual/issues/470). Included simple documentation until the Well spec has been completed in the Design Manual.

## Additions

- Added the Well organism to cf-layout.

## Changes

- Updated the cf-layout documentation to include the well example.

## Testing

- Follow the instructions [to test components locally](https://github.com/cfpb/capital-framework/blob/canary/CONTRIBUTING.md#testing-components-locally)
- run `gulp build`
- navigate to `http://localhost:3000/components/cf-layout/#wells`

## Screenshots

screen shot 2017-11-28 at 12 16 56 pm

## Notes

- Wells no longer span edge to edge like Featured Content Modules [per this decision](https://github.com/cfpb/design-manual/issues/469)

## Todos

- Update DM documentation with live examples.

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### Browsers

- [x] Chrome
- [x] Safari
- [x] FF
- [x] IE10
- [x] IE9
- [x] IE8
- [x] Opera
- [x] iOS
- [x] Android
- [x] Blackberry Bold

### Accessibility

- [x] Keyboard friendly
- [x] Screen reader friendly

### Other

- [x] Is useable without CSS
- [x] Is useable without JS
- [x] Flexible from small to large screens
- [x] No linting errors or warnings
- [x] JavaScript tests are passing

### CODE REVIEW

**Comment by nataliafitzgerald** [comment]
Looks good @jimmynotjim! Note to myself for the DM to max out the width of the well at 770px.

**Comment by Scotchester** [comment]
@nataliafitzgerald The width of the well, or the width of text inside it?

**Comment by nataliafitzgerald** [comment]
@Scotchester 
The max width of the well container would be 770px (regardless of page type) and the max width of the content in the well would be 670px (our standard). I'm working on mock-ups of images which I will share on the Design Manual side of things.

**Comment by jimmynotjim** [comment]
@nataliafitzgerald Updated for the max-width of 8 columns (770px).

screen shot 2017-11-30 at 3 01 18 pm

**Comment by nataliafitzgerald** [comment]
@jimmynotjim 
The width in your screenshot looks a little too wide. I recreated this in the Chrome browser and it looks like this: 
screen shot 2017-11-30 at 6 07 21 pm

Can you also post a mobile screenshot when you have a chance?

**Comment by jimmynotjim** [comment]
@nataliafitzgerald thanks for catching the width issue, I've fixed it and included small screen screenshot

screen shot 2017-12-01 at 11 02 59 am

screen shot 2017-12-01 at 11 03 25 am

**Comment by nataliafitzgerald** [comment]
@jimmynotjim - This looks great!

File: src/cf-layout/src/organisms/wells.less [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.o-well {
+    box-sizing: border-box;
+
+    // 8 columns plus gutters
+    max-width: unit( 770px / @base-font-size-px, em );
```
Comment (Scotchester) [inline src/cf-layout/src/organisms/wells.less]: Not a huge fan of this magic number. cf-layout already requires cf-grid, so we could calculate it based on cf-grid vars, right?

File: src/cf-layout/usage.md [hunk 2]
```diff
@@ -1989,3 +1990,29 @@ with a non-white (or light gray) background.
     </style>
 </section>
 ```
+
```
Comment (Scotchester) [inline src/cf-layout/usage.md]: Trivial: one more blank line here would be appreciated.

File: src/cf-layout/src/organisms/wells.less (hunk 1 above)
Comment (jimmynotjim) [reply src/cf-layout/src/organisms/wells.less]: Grid columns don't have a px value and we don't want to limit it to a % of the content area, just make sure it doesn't grow larger than 8 columns at max-width.

**Pull request merged**